File: parse_keys.js
import fs from 'fs';

const content = fs.readFileSync('src/lib/dynamicPathways.ts', 'utf8');

// A very naive regex to get top-level keys in the dynamicPathways object.
const match = content.match(/export const dynamicPathways: Record<string, DynamicPathway> = \{([\s\S]*?)\n\};/);
if (match) {
    const body = match[1];
    // console.log("Body length:", body.length);
    // write a simple parser to find the start and end of 'ispa': { ... }
    let ispaStart = content.indexOf(`'ispa': {`);
    if (ispaStart !== -1) {
        let braces = 0;
        let ispaEnd = -1;
        for (let i = ispaStart + 8; i < content.length; i++) {
            if (content[i] === '{') braces++;
            else if (content[i] === '}') {
                if (braces === 0) {
                    ispaEnd = i;
                    break;
                }
                braces--;
            }
        }
        
        if (ispaEnd !== -1) {
            // Remove the block
            // Also need to get right before 'ispa' to find the comment
            const commentStart = content.lastIndexOf('// ISPA (Infeksi Saluran Pernapasan Akut)', ispaStart);
            if (commentStart !== -1) {
                ispaStart = commentStart;
            }
            const newContent = content.substring(0, ispaStart) + "  'ispa': ispaPathway," + content.substring(ispaEnd + 1);
            fs.writeFileSync('src/lib/dynamicPathways.ts', newContent);
            console.log("Successfully replaced inline 'ispa' with 'ispaPathway'!");
        } else {
            console.log("Could not find matching brace");
        }
    } else {
        console.log("could not find 'ispa': {");
    }
}

File: parse_keys.py
import re

with open('src/lib/dynamicPathways.ts', 'r') as f:
    content = f.read()

ispa_idx = content.find("'ispa': {")
if ispa_idx != -1:
    comment_idx = content.rfind("// ISPA (Infeksi Saluran Pernapasan Akut)", 0, ispa_idx)
    if comment_idx != -1:
        start_idx = comment_idx
    else:
        start_idx = ispa_idx
        
    braces = 0
    end_idx = -1
    for i in range(ispa_idx + 8, len(content)):
        if content[i] == '{':
            braces += 1
        elif content[i] == '}':
            if braces == 0:
                end_idx = i
                break
            braces -= 1
            
    if end_idx != -1:
        new_content = content[:start_idx] + "  'ispa': ispaPathway," + content[end_idx+1:]
        with open('src/lib/dynamicPathways.ts', 'w') as f:
            f.write(new_content)
        print("Successfully replaced inline 'ispa' with 'ispaPathway'!")
    else:
        print("Could not find matching brace for 'ispa': {")
else:
    print("Could not find 'ispa': {")
